app/controllers/foodcoop/users_controller.rb

```diff
@@ -1,6 +1,25 @@
 class Foodcoop::UsersController < ApplicationController
   def index
-    @users = User.undeleted.natural_order
+    sort = if params["sort"]
+             case params["sort"]
+             when "name" then "first_name, last_name"
+             when "name_reverse" then "first_name DESC, last_name DESC"
+             when "email" then "email"
+             when "email_reverse" then "email DESC"
+             when "phone" then "phone"
+             when "phone_reverse" then "phone DESC"
+             when "ordergroup" then "groups.name"
+             when "ordergroup_reverse" then "groups.name DESC"
+             end
+           else
+             "first_name, last_name"
+           end
+
+    @users = case params["sort"]
+             when "ordergroup", "ordergroup_reverse" then User.left_joins(:groups).where("groups.type = 'Ordergroup'").undeleted.order(sort).distinct
+             else
+               User.undeleted.order(sort)
+             end
```

Comment:
please share at least the `.undeleted.order(sort)`. i'd just remove the whole `case` here and do the join always to keep the code simpler. also i'd suggest to avoid the strings in the `where` and use `where(groups:{type:'Ordergroup'})` instead

Reply:
Done!